copylot/hardware/cameras/flir/flir_camera.py

```diff
@@ -0,0 +1,484 @@
+        """
+        Initialize the current camera. Used for settings or to start imaging.
+        """
+        if not self.cam.IsInitialized():
+            self.cam.Init()
+
+    def close(self):
+        """
+        Close the system and delete pointer to current camera after imaging to avoid Spinnaker::Exception [-1004]
+        Call open() to image again with this instance
+        """
+        # Clean CameraPtr
+        self.cam = None
+        # Clear camera list
+        self.cam_list.Clear()
+        # Release system
+        self.system.ReleaseInstance()
+
+        # set back to default
+        self.system = None
+        self.cam_list = None
+        self.device_id = None
+        self.nodemap_tldevice = None
+
+    def list_available_cameras(self):
+        """
+        Return the list (type CameraList) of all the cameras in the system
+        """
+        return self.cam_list
+
+    def save_image(self, n, serial_no, processor, wait_time):
+        """
+        Save the nth image to take from a given, initialized camera
+
+        Parameters
+        ----------
+        n: number of images to take in that period of camera initialization. Type: int
+        serial_no: serial number of camera. Type: string
+        processor: image processor for post-processing. Type: ImageProcessor
+        wait_time: wait time for camera to take one frame in microseconds.
+        """
+        #  Retrieve next received image.
+        image_result = self.cam.GetNextImage(wait_time)
+        if image_result.IsIncomplete():
+            logger.warning(
+                'Image incomplete with image status %d ...'
+                % image_result.GetImageStatus()
+            )
+            return False
+        else:
+            #  some image information
+            width = image_result.GetWidth()
+            height = image_result.GetHeight()
+            logger.info(f"Image width, height: {width} {height}")
+
+            #  Convert image to mono 8. Converted has no need to be released. Optional color processing.
+            image_converted = processor.Convert(image_result, PySpin.PixelFormat_Mono8)
+            # Create a unique filename
+            if serial_no:
+                filename = 'Acquisition-%s-%d.jpg' % (serial_no, n)
+                while os.path.isfile(
+                    './' + filename
+                ):  # avoids overwriting if calling snap() multiple times
+                    n = n + 1
+                    filename = 'Acquisition-%s-%d.jpg' % (serial_no, n)
+            else:  # if serial number is empty
+                filename = 'Acquisition-%d.jpg' % n
+                while os.path.isfile('./' + filename):
+                    n = n + 1
+                    filename = 'Acquisition-%d.jpg' % (n + 1)
+            #  Save image
+            image_converted.Save(filename)
+            logger.info('Image saved at %s' % filename)
+            #  Release image
+            image_result.Release()
+            return True
+
```

Comment:
Could you also add the support for .tiff as well as ome-zarr ? maybe add a function that just grab the data from the camera and return an array and saving can be handle outside the camera adaptor ??

Reply:
@aduran-cz you can get support for both .tiff and ome-zarr from iohub as we discussed

Reply:
I added support for .tiff with skim age.io, but not for one-zarr because PySpin uses Python 3.8 and Biohub requires >= 3.9

Reply:
we will use pyspin  3.9 on windows machines(target env) so it won't be a problem to have iohub @aduran-cz but you can make a separate PR for it